src/utilities/result.ts

```diff
@@ -7,9 +7,8 @@ export interface FailureResult<Reason> extends BaseResult {
   readonly reason: Reason;
 }
 
-export interface SuccessfulResult<Result> extends BaseResult {
-  readonly didSucceed: true;
-  readonly result: Result;
-}
+export type SuccessfulResult<Result> = Result extends undefined
```

Comment:
Is this change really necessary? Couldn't we just do `Result<void, FailureReason>`?

If we did need this change, or a version of it, would we really need to do `BaseResult & ...` in the lines below? It seems redundant.